tinyxml2.h

```diff
@@ -789,7 +790,7 @@ class TINYXML2_LIB XMLNode
     const XMLElement* LastChildElement( const char* name = 0 ) const;
 
     XMLElement* LastChildElement( const char* name = 0 )	{
-        return const_cast<XMLElement*>(const_cast<const XMLNode*>(this)->LastChildElement(name) );
+        return const_cast<XMLElement*>(static_cast<const XMLNode*>(this)->LastChildElement(name) );

```

Comment:
This makes things less clear. `const_cast` only adds/removes CV qualifiers but `static_cast` can do much more than that. Using `const_cast` shows that adding `const` in the only magic that takes place here.

Reply:
Personally, I've only seen `const_cast` as having the real power removing `const` (and `volatile`). `static_cast` can also _add_ constness, but lacks the power of removal. IMHO, knowing that means `static_cast` is usually used for downcast only but also to add const.